src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -0,0 +1,344 @@
+import javax.annotation.Nullable;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.stellar.sdk.requests.ClientIdentificationInterceptor;
+import org.stellar.sdk.requests.ResponseHandler;
+import org.stellar.sdk.requests.sorobanrpc.GetEventsRequest;
+import org.stellar.sdk.requests.sorobanrpc.GetLedgerEntriesRequest;
+import org.stellar.sdk.requests.sorobanrpc.GetTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SendTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SimulateTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SorobanRpcErrorResponse;
+import org.stellar.sdk.requests.sorobanrpc.SorobanRpcRequest;
+import org.stellar.sdk.responses.sorobanrpc.GetEventsResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetHealthResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetLatestLedgerResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetLedgerEntriesResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetNetworkResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SendTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SimulateTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SorobanRpcResponse;
+import org.stellar.sdk.xdr.ContractDataDurability;
+import org.stellar.sdk.xdr.ContractEntryBodyType;
+import org.stellar.sdk.xdr.LedgerEntry;
+import org.stellar.sdk.xdr.LedgerEntryType;
+import org.stellar.sdk.xdr.LedgerKey;
+import org.stellar.sdk.xdr.SCVal;
+import org.stellar.sdk.xdr.SorobanAuthorizationEntry;
+import org.stellar.sdk.xdr.XdrDataInputStream;
+import org.stellar.sdk.xdr.XdrDataOutputStream;
+
+/**
+ * Main class used to connect to the Soroban-RPC instance and exposes an interface for requests to
+ * that instance.
+ */
+public class SorobanServer implements Closeable {
+  private static final int SUBMIT_TRANSACTION_TIMEOUT = 60; // seconds
+  private static final int CONNECT_TIMEOUT = 10; // seconds
+  private final HttpUrl serverURI;
+  private final OkHttpClient httpClient;
+  private final Gson gson = new Gson();
+
+  /**
+   * Creates a new SorobanServer instance.
+   *
+   * @param serverURI The URI of the Soroban-RPC instance to connect to.
+   */
+  public SorobanServer(String serverURI) {
+    this(
+        serverURI,
+        new OkHttpClient.Builder()
+            .addInterceptor(new ClientIdentificationInterceptor())
+            .connectTimeout(CONNECT_TIMEOUT, TimeUnit.SECONDS)
+            .readTimeout(SUBMIT_TRANSACTION_TIMEOUT, TimeUnit.SECONDS)
+            .retryOnConnectionFailure(true)
+            .build());
+  }
+
+  public SorobanServer(String serverURI, OkHttpClient httpClient) {
+    this.serverURI = HttpUrl.parse(serverURI);
+    this.httpClient = httpClient;
+  }
+
+  public TransactionBuilderAccount getAccount(String accountId)
```

Comment:
would be helpful to javadoc the public methods, both for readability of source and also for client usage.

Reply:
Fixed.